src/server/application/lib/Repository.ts

```diff
@@ -0,0 +1,12 @@
+import { Entity } from '../../core/lib'
+
+export interface Repository<T extends Entity> {
+  add(entity: T): Promise<void>
+  find(id: T['id']): Promise<T | null>
+  findAll(): Promise<T[]>
+  getLast(): Promise<T | null>
+  has(id: T['id']): Promise<boolean>
+  remove(id: T['id']): Promise<boolean>
+  size(): Promise<number>
+  update(id: T['id'], entity: T): Promise<void>
```

Comment:
If the type requires the entire entity, can this function signature be simplified to:

```
update(entity: T): Promise<void>
```

Alternatively, if a partial entity can be passed to update, I'd suggest this signature

```
update(id: T['id'], entity: Partial<T>): Promise<void>
```

Reply:
Good catch. I've simplified the signature as I'm only considering full entities to be passed at this stage